### PR TITLE
Also make a faster DotproductExecutor when using enumerated values. MERGEOK

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/features/dotproductfeature.cpp [hunk 1]
```diff
@@ -124,6 +124,48 @@ DotProductExecutor<A>::getAttributeValues(uint32_t docId, const AT * & values)
...
+
+DotProductExecutorByEnum::~DotProductExecutorByEnum() = default;
+
+void DotProductExecutorByEnum::execute(uint32_t docId) {
+    feature_t val = 0;
+    if (!_queryVector.getDimMap().empty()) {
+        const IWeightedIndexVector::WeightedIndex *values(nullptr);
+        uint32_t sz = _attribute->getEnumHandles(docId, values);
+        for (size_t i = 0; i < sz; ++i) {
+            typename V::HashMap::const_iterator itr = _queryVector.getDimMap().find(values[i].value().ref());
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/dotproductfeature.cpp]: Consider using auto.

File: build_settings.cmake [hunk 2]
```diff
@@ -28,7 +28,7 @@ else()
 endif()
 
 # C and C++ compiler flags
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -O3 -fno-omit-frame-pointer ${C_WARN_OPTS} -fPIC ${VESPA_CXX_ABI_FLAGS} -DBOOST_DISABLE_ASSERTS ${VESPA_CPU_ARCH_FLAGS} -mtune=intel ${EXTRA_C_FLAGS}")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -O0 -fno-omit-frame-pointer ${C_WARN_OPTS} -fPIC ${VESPA_CXX_ABI_FLAGS} -DBOOST_DISABLE_ASSERTS ${VESPA_CPU_ARCH_FLAGS} -mtune=intel ${EXTRA_C_FLAGS}")
```
Comment (geirst) [inline build_settings.cmake]: Please change back to -O3.

**Review by geirst — CHANGES_REQUESTED** [review]
Overall looks good.

**Review by geirst — APPROVED** [review]

File: searchlib/src/vespa/searchlib/features/dotproductfeature.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/features/dotproductfeature.cpp]: Fixed

File: build_settings.cmake (hunk 2 above)
Comment (baldersheim) [reply build_settings.cmake]: Done

**Pull request merged**